QCoroMacros.cmake

```diff
@@ -4,7 +4,7 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fcoroutines")
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fcoroutines-ts -stdlib=libc++")
-else()
+elseif (NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```

Comment:
According to https://docs.microsoft.com/en-us/cpp/build/reference/await-enable-coroutine-support?view=msvc-160 you need to set `/await:strict` option to enable C++20 coroutines support in MSVC.

It might be possible to enable it through VisualStudio (not sure), but I would prefer to have it explictly set in CMake.

Reply:
After specifying C++20 in cmake, cmake will passes `/std:c++latest` to MSVC, which enables coroutine by default.
Actually MSVC supports both C++20 standard coroutine (`#include <coroutine>`) and C++17 with coroutine TS (`#include <experimental/coroutine>`). I think we don't need to support the latter one. If users need to use coroutine TS, they can change the compiler flags.

Reply:
OK, I wasn't aware of that. Thanks for the explanation.